src/index.js

```diff
@@ -1,37 +1,49 @@
-function log(prevState, action, nextState) {
-  console.group('%c action', 'color: gray; font-weight: lighter;', action.name);
-  console.log('%c prev state', 'color: #9E9E9E; font-weight: bold;', prevState);
-  console.log('%c data', 'color: #03A9F4; font-weight: bold;', action.data);
-  console.log('%c next state', 'color: #4CAF50; font-weight: bold;', nextState);
-  console.groupEnd();
+function defaultLog(prevState, action, nextState) {
+  console.group("%c action", "color: gray; font-weight: lighter;", action.name)
+  console.log("%c prev state", "color: #9E9E9E; font-weight: bold;", prevState)
+  console.log("%c data", "color: #03A9F4; font-weight: bold;", action.data)
+  console.log("%c next state", "color: #4CAF50; font-weight: bold;", nextState)
+  console.groupEnd()
 }
 
 export default function(options) {
-  options = options || {};
-  options.log = typeof options.log === 'function' ? options.log : log;
+  options = options || {}
+  options.log = typeof options.log === "function" ? options.log : defaultLog
 
-  return function(emit) {
-    var actionStack = [];
-    return {
-      events: {
-        action: function(state, actions, action) {
-          actionStack.push(action);
-        },
-        resolve: function(state, actions, result) {
-          if (typeof result === 'function') {
-            var action = actionStack.pop();
-            return function(update) {
-              return result(function(result) {
-                actionStack.push(action);
-                return update(result);
-              });
-            };
-          }
-        },
-        update: function(state, actions, nextState) {
-          options.log(state, actionStack.pop(), nextState);
-        }
+  return function(app) {
+    return function(props) {
+      function enhance(actions) {
+        return Object.keys(actions).reduce(function(otherActions, name) {
+          const action = actions[name]
+          otherActions[name] =
+            typeof action === "function"
+              ? function(state, actions, data) {
+                  const result = action(state, actions, data)
+                  if (typeof result === "function") {
+                    return function(update) {
+                      return result(function(withState) {
+                        options.log(
+                          state,
+                          { name: name, data: data },
+                          withState
+                        )
+                        return update(withState)
+                      })
+                    }
+                  } else {
+                    options.log(state, { name: name, data: data }, result)
+                    return result
```

Comment:
@okwolf This implementation only logs the state fragment, but not the global state, which is likely what most people would want.

Reply:
@JorgeBucaran is that a bug, or a feature? Actions that are scoped to state slices are **only** able to update that slice, and including any other state might actually be seen as superfluous noise in analyzing the `state` diffs 🤔

Reply:
@okwolf Sure, and that's fine, so perhaps the way to go: 

```
sliceState, actionInfo, actionResult, globalState
```

Reply:
@JorgeBucaran is `globalState` supposed to be the updated/after `state`? Because at this point the update hasn't happened yet and calling a get type action will still have the global before state.